src/core/gosling.schema.guards.ts

```diff
@@ -211,17 +219,43 @@ export function IsStackedChannel(track: SingleTrack, channelKey: keyof typeof Ch
  * Retreive value using a `channel`.
  * `undefined` if unable to retreive the value.
  */
-export function getValueUsingChannel(datum: { [k: string]: string | number }, channel: Channel) {
-    if (IsChannelDeep(channel) && channel.field) {
-        return datum[channel?.field];
+export function getValueUsingChannel(
+    datum: { [k: string]: string | number },
+    channel: Channel,
+    fieldKey?: 'field' | 'startField' | 'endField' | 'startField2' | 'endField2'
+) {
+    const field = getChannelField(channel, fieldKey);
+    if (IsChannelDeep(channel) && field) {
+        return datum[field];
+    }
+    return undefined;
+}
+
+/**
+ * Get a field name from a channel. If there are multiple fields (e.g., start and end fields), get the first one
+ * unless `channelKey` is defined. Return `undefined` if a channel does not have any fields specified.
+ */
+export function getChannelField(
+    channel: Channel | undefined,
+    fieldKey?: 'field' | 'startField' | 'endField' | 'startField2' | 'endField2'
+): string | undefined {
+    if (IsMultiFieldChannel(channel)) {
+        // This means a channel is either X or Y and contains multiple fields (e.g., start and end fields).
+        return channel[!fieldKey || fieldKey === 'field' ? 'startField' : fieldKey];
+    } else if (channel && 'field' in channel) {
+        // This means a channel contains a single field
+        if (!fieldKey || fieldKey === 'field') {
+            return channel.field;
+        }
     }
+    // Otherwise, just return `undefined`
     return undefined;
 }
 
 export function getChannelKeysByAggregateFnc(spec: SingleTrack) {
     const keys: (keyof typeof ChannelTypes)[] = [];
     SUPPORTED_CHANNELS.forEach(k => {
-        const c = spec[k];
+        const c = spec.encoding[k];
```

Comment:
Aggregating `startField` and `endField` separately should be still possible even when we group `x` and `xe`.